components/composed/model/ModelList/hooks/useRowActions.tsx

```diff
@@ -48,14 +53,25 @@ function getButtonProps<D extends Record<string, unknown>>(
 ) {
   const actionDefinition = availableActions[action];
 
-  const buttonProps = {
-    "aria-label": actionDefinition.label,
-    icon: actionDefinition.icon,
-    iconRotate: actionDefinition.iconRotate || 0,
-    ...(actionConfig?.handleClick && {
-      onClick: () => actionConfig.handleClick({ row }),
-    }),
-  };
+  const buttonProps = actionConfig?.modalConfirm
+    ? {
+        "aria-label": actionDefinition.label,
+        icon: actionDefinition.icon,
+        iconRotate: actionDefinition.iconRotate || 0,
+        ...(actionConfig?.handleClick && {
+          onClick: () => actionConfig.handleClick({ row }),
+        }),
+        modalLabel: actionDefinition.modalLabel ?? null,
+        modalBody: actionDefinition.modalBody ?? null,
+      }
+    : {
+        "aria-label": actionDefinition.label,
+        icon: actionDefinition.icon,
+        iconRotate: actionDefinition.iconRotate || 0,
+        ...(actionConfig?.handleClick && {
+          onClick: () => actionConfig.handleClick({ row }),
+        }),
+      };
```

Comment:
@jen-castiron We should look at this in conjunction with the issue of simplifying `ButtonControl`. It would probably make sense to just pass a `drawer` or `modal` flag as part of the action in `useRowActions` and then instead of making these prop objects, just pass the correct `ButtonControlGroup` child here like you propose doing for button controls in the list headers. I've modeled the pattern for the flag in the action for modal, but we'd have to also implement it for drawer.